src/app/content/practiceQuestions/reducer.ts

```diff
@@ -29,7 +30,7 @@ const reducer: Reducer<State, AnyAction> = (state = initialState, action): State
     case getType(receiveFeatureFlags):
       return {...state, isEnabled: action.payload.includes(practiceQuestionsFeatureFlag)};
     case getType(actions.receivePracticeQuestionsSummary):
-      return {...state, summary: action.payload};
+      return {...state, loading: true, summary: action.payload};
```

Comment:
shouldn't this be set on `locationChange` since we dispatch `receivePracticeQuestionsSummary` only once per book and the screen depends on `questionsCount` selector which depends on `setQuestions`?

There is also `setSelectedSection` hook.
I think we should set loading before we call
```
await practiceQuestionsLoader.getPracticeQuestions
```

It may not be usefull right now since we are getting these questions from prefetched jsons, but the intention of `practiceQuestionsLoader` is that we shouldn't expect anything from it

Reply:
There are 3 different actions scenarios:
1. Open modal directly via modal-ulr
`location change` -> `receivePracticeQuestionsSummary` -> `setSelectedSection` -> `setQuestions`
2. Open modal with button
`location change` -> `setSelectedSection` -> `setQuestions`
3. New section with question (Intro or Final)
`setSelectedSection` -> `setQuestions`

Setting `loading: true;` for `location change` and `setSelectedSection` would cover all scenarios, but it gives a little redundancy.

Reply:
There will be 4th scenario when we implement #941 - selectin a section from a chapter filter
I think all the cases have call to `setSelectedSection` so I would go with `loading: true;` for `setSelectedSection` and then `loading: false` for `setQuestions`
Will this cause delay in the first scenario?

Reply:
> There will be 4th scenario when we implement #941 - selecting a section from a chapter filter

Sounds similar to 3rd scenario in terms of actions.
> Will this cause delay in the first scenario?

It would cause delay in 1st and 2nd scenario.

Reply:
right, it's basically the same
What about `loading: true` for both `setSelectedSection` and `locationChange` and `loading: false` for `setQuestions`?

Reply:
It handles all scenarios.